python/example_code/ses/ses_replicateidentities/ses_replicateidentities.py

```diff
@@ -0,0 +1,343 @@
+                                          HostedZoneId=zone_id,
+                                          ChangeBatch=batch
+                                          )
+                                # print(add_txt)
+                            except ClientError as err:
+                                print (err)
+                                if (err.response['Error']['Code'] ==
+                                     'InvalidChangeBatch'):
+                                    print ("Check if TXT record",
+                                           "for the domain already exists.\n")
+                        else:
+                            pass
+
+                    if is_txt_present is False:
+                         token = json.dumps(table[domain])
+                         batch = {
+                                  "Changes": [
+                                              {"Action": "UPSERT",
+                                              "ResourceRecordSet":
+                                               {"Name": ses_prefix+domain,
+                                                "Type": "TXT", "TTL": 1800,
+                                                "ResourceRecords":
+                                                [{"Value": token}]}
+                                               }
+                                              ]
+                                  }
+                         add_txt = r53.change_resource_record_sets(
+                                   HostedZoneId=zone_id,
+                                   ChangeBatch=batch
+                                   )
+                else:
+                    pass
+    elif rec_type == 'dkimVerify':
+        for zone in zone_list:
+            compare = '.'.join(dkim_dom.split('.')[-2:])
+            print(compare)
+            if compare == zone['Name'][:-1]:
+                zone_id = zone['Id'].strip('/hostedzone/')
+                batch = {
+                        "Changes": [
+                                    {"Action": "UPSERT",
+                                     "ResourceRecordSet":
+                                     {"Name": dkim_tok+"._domainkey."+dkim_dom,
+                                      "Type": "CNAME", "TTL": 1800,
+                                      "ResourceRecords":
+                                      [{"Value": dkim_tok+".dkim.amazonses.com"}
+                                       ]
+                                      }
+                                     }
+                                    ]
+                         }
+                add_txt = r53.change_resource_record_sets(
+                                                          HostedZoneId=zone_id,
+                                                          ChangeBatch=batch
+                                                         )
+                print(add_txt)
+            else:
+                pass
```

Comment:
Refactor this code so it doesn't have so many indented sub-clauses. Perhaps pull some parts of it into other functions?

Reply:
The responses returned by the API calls done within this function and their processing is such that I would rather it remain a single block of logic.